string msg
float64 angle
int32 circle_state
bool fast_axis
bool apply_config

